test/tests/test_optional_argument.cpp

```diff
@@ -139,6 +139,62 @@ TEST_CASE_FIXTURE(
     REQUIRE_FALSE(sut_has_value(sut));
 }
 
+TEST_CASE_FIXTURE(
+    optional_argument_test_fixture,
+    "choices(any) should throw when passed choices are not of test_value_type"
+) {
+    auto sut = prepare_argument(long_name);
+    std::vector<std::any> choices = {"1", "2", "3"};
+
+    REQUIRE_THROWS_AS(sut_set_choices(sut, choices), std::invalid_argument);
+}
+
+TEST_CASE_FIXTURE(
+    optional_argument_test_fixture,
+    "value(any) should accept the arguement's value if it matches an entry set using choices()"
```

Comment:
"value(any) should accept the given value only when it's present in the choices set"